File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        count = 0
        for ch in s:
            if t.find(ch, count) == -1:
                return False
            else:
                count = t.find(ch, count) + 1

        return True

File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int p = 0;
        int i;
        for (i = 0; i < s.length() && p < t.length(); i++) {
            while(p < t.length() && s.charAt(i) != t.charAt(p)) {
                p++;
            }
            if (p == t.length() || s.charAt(i) != t.charAt(p)) {
                return false;
            }
            // System.out.println(s.charAt(i) + " found at position " + p);
            p++;
        }
        if (i == s.length()) {
            return true;
        }
        return false;
    }
}